src/render/opengles2/SDL_render_gles2.c

```diff
@@ -543,20 +543,19 @@ static GLuint GLES2_CacheShader(GLES2_RenderData *data, GLES2_ShaderType type, G
     }
 
     if (!compileSuccessful) {
-        SDL_bool isstack = SDL_FALSE;
         char *info = NULL;
         int length = 0;
 
         data->glGetShaderiv(id, GL_INFO_LOG_LENGTH, &length);
         if (length > 0) {
-            info = SDL_small_alloc(char, length, &isstack);
+            info = (char *)SDL_malloc(length);
             if (info) {
                 data->glGetShaderInfoLog(id, length, &length, info);
             }
         }
         if (info) {
             SDL_SetError("Failed to load the shader %d: %s", type, info);
-            SDL_small_free(info, isstack);
+            SDL_free(info);
         } else {
```

Comment:
I don't fully understand the warning here, but: does the warning go away if you move this `if (info)` block to above?

Reply:
No, the warning is complaining because it might use a stack allocation with too large a size. This code isn't performance critical, so it's fine to always use a heap allocation.